File: packages/expo-speech/android/src/main/java/expo/modules/speech/SpeechOptions.kt
package expo.modules.speech

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record
import expo.modules.kotlin.types.OptimizedRecord

@OptimizedRecord
data class SpeechOptions(
  @Field val language: String?,
  @Field val pitch: Float?,
  @Field val rate: Float?,
  @Field val voice: String?,
  @Field val volume: Float?
) : Record

File: packages/expo-speech/ios/SpeechOptions.swift
import ExpoModulesCore

struct SpeechOptions: Record {
  @Field
  var language: String?
  @Field
  var pitch: Double?
  @Field
  var rate: Double?
  @Field
  var voice: String?
}

struct VoiceInfo: Record {
  @Field
  var identifier: String? = nil
  @Field
  var name: String? = nil
  @Field
  var quality: String? = nil
  @Field
  var language: String? = nil
}
